src/reader/mod.rs

```diff
@@ -107,11 +112,27 @@ impl IndexReaderBuilder {
         self
     }
 
-    /// Sets the number of `Searcher` in the searcher pool.
+    /// Sets the number of [Searcher] to pool.
+    ///
+    /// When more than `num_searchers` are requested, the caller will block.
```

Comment:
```suggestion
    /// See [Self::searcher()].
```
This might seem like a weird suggestion. The documentation seemed to suggest that the called to `num_searchers()` would block.